components/inputs/input-number.js

```diff
@@ -63,14 +89,24 @@ class InputNumber extends LitElement {
 				class="d2l-input"
 				?disabled="${this.disabled}"
 				id="${this._inputId}"
-				max="${ifDefined(this.max)}"
-				min="${ifDefined(this.min)}"
 				name="${ifDefined(this.name)}"
 				placeholder="${ifDefined(this.placeholder)}"
-				step="${ifDefined(this.step)}"
-				type="number"
-				.value="${this.value}">
+				type="text"
+				.value="${this._formattedValue}">
 		`;
 	}
+
+	_handleChange(e) {
+		const inputValue = e.target.value.trim();
+		this.value = inputValue !== '' ? parseNumber(inputValue) : undefined;
+
+		if (!isNaN(this.value) && this.value !== undefined) {
+			this.value = checkMinMaxValue(this.value, this.min, this.max);
+			this._formattedValue = formatValue(this.value, this.minFractionDigits, this.maxFractionDigits);
+			e.target.value = this._formattedValue;
```

Comment:
You never want to be setting `e.target.value`, since it's already bound to `this._formattedValue`. Simply setting `this._formattedValue` should be enough to trigger a re-render, which should set the value of the input.

Similarly, with the changes called out above, I don't think you even need to set `_formattedValue` since it should be a function of setting value.

Reply:
See the discussion [here](https://github.com/BrightspaceUI/core/pull/788#discussion_r475856822) about `_formattedValue` being a function of `set value()`